shared/utils/prismic.js

```diff
@@ -14,67 +15,74 @@ function linkResolver(doc) {
     case 'articles':
       return '/articles';
     case 'article':
-      return `/articles/${doc.slug || doc.uid}`;
+      return `/articles/${doc.uid}`;
     case 'custom_page':
-      return `/${doc.slug || doc.uid}`;
+      return `/${doc.uid}`;
     default:
       return '/';
   }
 }
 
-function renderAsText(field) {
-  try {
-    return asText(field);
-  } catch (e) {
-    console.warn('unable to render field as text', field);
-  }
+const SHOULD_WARN = true;
 
-  return '';
+function warn(...m) {
+  if (SHOULD_WARN) {
+    console.warn(...m);
+  }
 }
 
-function renderAsComponentTree(field) {
-  if (!field) {
-    console.warn('cannot render undefined field');
-    return null;
+function get(obj, path = undefined, defaultValue = '') {
+  const value = path ? _get(obj, path) : obj;
```

Comment:
Path should be required, the same way lodash's _.get works. Otherwise this can sometimes return the whole object - which will print `[Object object]`

Reply:
Can you rename this to `getString()` and always ensure a string is returned?

Reply:
I wanted it to be flexible and allow for non-objects also, i.e. if you've somehow got the value from the prismic object and want a string value for it, like here
https://github.com/ueno-llc/demo-prismic/blob/master/shared/MainApp.js#L45

```
const customPages = jobResult.data.custom_pages
  .map(({ custom_page: { uid, data: { title } } }) => (
    <Link key={uid} to={`/${uid}`}>{getString(title)}</Link>
  ));
```

or here
https://github.com/ueno-llc/demo-prismic/blob/master/shared/routes/about/About.js#L43

```
{people.map(({ person: { data: { name, bio, image } } }, i) => (
  <People
    key={`people-${i}`} // eslint-disable-line
    image={getObject(image).url}
    name={getString(name)}
    description={getString(bio)}
  />
))}
```

but looking more closely this should be refactored since it's fragile to nulls

Reply:
If it's called `getString`, you should only return `obj` if you can determine that `typeof obj === 'string'`